src/gsi/control_vectors.f90

```diff
@@ -895,7 +895,7 @@ real(r_quad) function qdot_prod_sub(xcv,ycv)
      m3d=xcv%step(1)%n3d
      m2d=xcv%step(1)%n2d
      itot=max(m3d,0)+max(m2d,0)
-     if(l_hyb_ens)itot=itot+n_ens
+     if(l_hyb_ens)itot=itot+n_ens*naensgrp
```

Comment:
Only this part is the modification to solve the issue of not working the option of scale/variable/time-dependent localization with the current trunk, right? (This is just confirmation.)

Reply:
I think so.  But when running in debug mode, there were several small changes that were necessary to get to this point in the code.  Better to fix those issues as well.